common/web/JsonPatchParser.cpp

```diff
@@ -73,15 +73,16 @@ void JsonPatchParser::String(const string &value) {
   switch (m_state) {
     case TOP:
       SetError(kPatchListError);
-      return;
+      break;
     case PATCH_LIST:
       SetError(kPatchElementError);
-      return;
+      break;
     case PATCH:
       HandlePatchString(value);
-      return;
-    default:
```

Comment:
No, we rely on the fact that m_state is an enum and the compiler will complain if we introduce another enum value. 

enumeration value 'X' not handled in switch